### ISSUE TITLE
What do we want from TravisCI?

### ISSUE THREAD

**Issue by mnlevy1981** [marbl-ecosys/MARBL#286]
Currently, [TravisCI](https://travis-ci.org/marbl-ecosys/MARBL) runs the following tests:

```
  - cd MARBL_tools; ./yaml_to_json.py
  - ./MARBL_generate_settings_file.py
  - ./MARBL_generate_diagnostics_file.py
  - cd ../tests/bld_tests; ./bld_exe.py
  - cd ../unit_tests/get_put; ./get_put.py
  - cd ../utils_routines; ./marbl_utils.py
  - cd ../../regression_tests/init; ./init.py
  - cd ../init-twice; ./init-twice.py
  - cd ../gen_input_file; ./gen_input_file.py
  - cd ../requested_diags; ./requested_diags.py
  - cd ../requested_forcings; ./requested_forcings.py
  - cd ../requested_restoring; ./requested_restoring.py
  - cd ../requested_tracers; ./requested_tracers.py
```

I was thinking that we can replace this list with a single script (maybe `MARBL_tools/run_test_suite.sh`) that can provide a cleaner summary about what tests passed and what tests failed. It would also let us look into adding baseline comparisons for some of the tests.

Additionally, do we want TravisCI to run the sphinx build?
* Pro: if a PR breaks the documentation, we would notice before accepting
* Con: we need to make sure that TravisCI uses the same python environment as whoever generates the documentation (which depends on whether we use ReadtheDocs or host the documentation on github-pages)

**Comment by matt-long** [comment]
I think it would be nice to have the docs built with every PR.  The environment issues seem like something we have to confront in any case.  Given that we've got the gh-pages approach to doc hosting down, I suggest we plan on sticking with that rather than thinking to migrate back to RTD—unless there are substantial advantages to do so.

**Comment by mnlevy1981** [comment]
closed via #287
